lib/OnyxUtils.js

```diff
@@ -1086,20 +1092,23 @@ function removeNullValues(key, value) {
  * to an array of key-value pairs in the above format and removes key-value pairs that are being set to null
  * @private
  * @param {Record} data
+ * @param {Boolean} shouldRemoveNestedNulls
  * @return {Array} an array of key - value pairs <[key, value]>
  */
-function prepareKeyValuePairsForStorage(data) {
-    const keyValuePairs = [];
-
-    _.forEach(data, (value, key) => {
-        const {value: valueAfterRemoving, wasRemoved} = removeNullValues(key, value);
-
-        if (wasRemoved) return;
+function prepareKeyValuePairsForStorage(data, shouldRemoveNestedNulls) {
+    return _.reduce(
+        data,
+        (pairs, value, key) => {
+            const {value: valueAfterRemoving, wasRemoved} = removeNullValues(key, value, shouldRemoveNestedNulls);
 
-        keyValuePairs.push([key, valueAfterRemoving]);
-    });
+            if (!wasRemoved) {
+                pairs.push([key, valueAfterRemoving]);
+            }
```

Comment:
NAB, we can return early if wasRemoved is true.

```js
if (wasRemoved) { return pairs; }
return pairs.push([key, valueAfterRemoving]);
```

Reply:
updated!